cnd/tests/ethereum_pattern.rs

```diff
@@ -30,3 +31,125 @@ fn cannot_skip_block_containing_transaction_with_event() {
 
     assert_that!(pattern.needs_receipts(&block)).is_false();
 }
+
+fn pattern_matches_block(pattern: TransactionPattern) -> bool {
+    let block: Block<Transaction> = include_json_test_data!("./test_data/ethereum/block.json");
+
+    for transaction in block.transactions.into_iter() {
+        if pattern.matches(&transaction, None) {
+            return true;
+        }
+    }
+    false
+}
+
+// `matches()` method is a filter i.e., it returns true unless things
+// specifically do _not_ match.  In order to _really_ test that the we
+// succesfully match we first do a negative test then do an identical positive
+// test.
+
+#[test]
+fn matches_transaction_from_address() {
+    let invalid_from_address = Address::from_str("fffffa1fba5b4804863131145bc27256d3abffff")
+        .expect("failed to construct from_address");
+
+    let pattern = TransactionPattern {
+        from_address: Some(invalid_from_address),
+        ..TransactionPattern::default()
+    };
+
+    let result = pattern_matches_block(pattern);
+    assert_that!(&result).is_false();
```

Comment:
🔧 You can split these up into two tests to make these more explicit and the tests follow the pattern: `Arrange/Act/Assert`. E.g.

```rust

#[test]
fn given_invalid_address_does_not_match_transaction_pattern() {
    let invalid_from_address = Address::from_str("fffffa1fba5b4804863131145bc27256d3abffff")
        .expect("failed to construct from_address");

    let pattern = TransactionPattern {
        from_address: Some(invalid_from_address),
        ..TransactionPattern::default()
    };

    let result = pattern_matches_block(pattern);
    assert_that!(&result).is_false();
}
```

and 

```rust

#[test]
fn given_valid_address_does_match_transaction_pattern() {
    let valid_from_address = Address::from_str("fb303a1fba5b4804863131145bc27256d3ab6692")
        .expect("failed to construct from_address");

    let pattern = TransactionPattern {
        from_address: Some(valid_from_address),
        ..TransactionPattern::default()
    };

    let result = pattern_matches_block(pattern);
    assert_that!(&result).is_true();
}
```


same applies for the other tests below.

Reply:
Cool, done!